runtime/lib/ttnn/runtime.cpp

```diff
@@ -29,39 +29,69 @@ static ::tt::target::Dim2d toFlatbuffer(CoreCoord coreCoord) {
   return ::tt::target::Dim2d(coreCoord.y, coreCoord.x);
 }
 
-std::pair<SystemDesc, DeviceIds> getCurrentSystemDesc() {
-  auto &device = ::ttnn::open_device(0);
-  std::vector<int> chipIds = {
-      device.id(),
-  };
+static std::pair<int, int> getConnectedChips(const ::ttnn::Device &device) {
+  std::unordered_set<chip_id_t> connectedChips =
+      device.get_ethernet_connected_device_ids();
+  std::vector<chip_id_t> connectedChipsOrdered(connectedChips.begin(),
+                                               connectedChips.end());
+  std::sort(connectedChipsOrdered.begin(), connectedChipsOrdered.end());
+  std::vector<int> connectedChipsWithDefault = {-1, -1};
+  for (size_t i = 0; i < connectedChipsOrdered.size(); i++) {
+    connectedChipsWithDefault[i] = connectedChipsOrdered[i];
+  }
+  return std::make_pair(connectedChipsWithDefault[0],
+                        connectedChipsWithDefault[1]);
+}
+
+static std::unique_ptr<std::pair<SystemDesc, DeviceIds>>
+getCurrentSystemDescImpl(std::vector<::ttnn::Device *> &mmio_devices,
+                         std::vector<::ttnn::Device *> &remote_devices) {
+  size_t numDevices = ::tt::tt_metal::GetNumAvailableDevices();
+  std::vector<int> chipIds;
+  std::vector<::flatbuffers::Offset<tt::target::ChipDesc>> chipDescs;
+  std::vector<uint32_t> chipDescIndices;
+  std::vector<::tt::target::ChipCapability> chipCapabilities;
+  std::vector<::tt::target::ChipCoord> chipCoords;
+  std::vector<::tt::target::ChipChannel> chipChannels;
   ::flatbuffers::FlatBufferBuilder fbb;
+
+  for (size_t deviceId = 0; deviceId < numDevices; deviceId++) {
+    ::ttnn::Device &device = ::ttnn::open_device(deviceId);
+    chipIds.push_back(device.id());
+    // Construct chip descriptor
+    ::tt::target::Dim2d deviceGrid = toFlatbuffer(device.logical_grid_size());
+    chipDescs.push_back(::tt::target::CreateChipDesc(
+        fbb, toFlatbuffer(device.arch()), &deviceGrid,
+        device.l1_size_per_core(), device.num_dram_channels(),
+        device.dram_size_per_channel(), L1_ALIGNMENT, PCIE_ALIGNMENT,
+        DRAM_ALIGNMENT));
+    chipDescIndices.push_back(deviceId);
+    // Derive chip capability
+    ::tt::target::ChipCapability chipCapability =
+        ::tt::target::ChipCapability::NONE;
+    if (device.is_mmio_capable()) {
+      chipCapability = chipCapability | ::tt::target::ChipCapability::PCIE |
+                       ::tt::target::ChipCapability::HostMMIO;
+      mmio_devices.push_back(&device);
+    } else {
+      remote_devices.push_back(&device);
+    }
+    chipCapabilities.push_back(chipCapability);
+    // Extract chip location
+    int x, y, rack, shelf;
+    std::tie(x, y, rack, shelf) = device.get_chip_location();
```

Comment:
Is `get_chip_location` the only new API you added to metal? If so let's file an issue on metal (to have an issue number in the commit) something like "Add device chip location API".

Reply:
Cool sounds good

Reply:
@nsmithtt Metal PR here: https://github.com/tenstorrent/tt-metal/pull/10674/files

issue here: https://github.com/tenstorrent/tt-metal/issues/10671